python/ray/libraylet_library_python.pyx

```diff
@@ -0,0 +1,723 @@
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from ray.includes.common cimport *
+from ray.includes.libraylet_client cimport LibRayletClient, ResourceMappingType, WaitResultPair
+from ray.includes.task cimport RayletTaskSpecification, RayletTaskExecutionSpecification, RayletTask, _Task, RayletTaskArgument, RayletTaskArgumentByValue, RayletTaskArgumentByReference
+from ray.includes.ray_config cimport RayConfig
+
+from libc.stdint cimport int32_t as c_int32, int64_t as c_int64
+from libcpp cimport bool as c_bool
+from libcpp.memory cimport shared_ptr, unique_ptr, make_shared, static_pointer_cast
+from libcpp.vector cimport vector as c_vector
+
+from libcpp.unordered_map cimport unordered_map
+from libcpp.utility cimport pair
+
+from cython.operator import dereference, postincrement
+cimport cpython
+
+if cpython.PY_MAJOR_VERSION >= 3:
+    import pickle
+else:
+    import cPickle as pickle
+import numpy
+
+
+include "includes/unique_ids.pxi"
+
+
+cdef c_vector[ObjectID] ObjectIDsToVector(object_ids):
+    """A helper function that converts a Python list of object ids to a vector.
+    
+    Args:
+        object_ids(list): The Python list of object ids.
+
+    Returns:
+        The output vector.
+    """
+    cdef:
+        PyObjectID object_id
+        c_vector[ObjectID] object_id_vector
+    for object_id in object_ids:
+        object_id_vector.push_back(object_id.data)
+    return object_id_vector
+
+
+cdef VectorToObjectIDs(c_vector[ObjectID] object_id_vector):
+    object_ids = []
+    for i in range(object_id_vector.size()):
+        object_ids.append(PyObjectID.from_native(object_id_vector[i]))
+    return object_ids
+
+
+def compute_put_id(PyTaskID task_id, c_int64 put_index):
+    if put_index < 1 or put_index > kMaxTaskPuts:
+        raise ValueError("The range of 'put_index' should be [1, %d]" % kMaxTaskPuts)
+    cdef CObjectID object_id = ComputePutId(task_id.data, put_index)
+    return PyObjectID.from_native(object_id)
+
+
+def compute_task_id(PyObjectID object_id):
+    return PyTaskID.from_native(ComputeTaskId(object_id.data))
+
+
+cdef c_bool is_simple_value(value, int *num_elements_contained):
```

Comment:
why not cdef int num_elements_contained at the beginning and at the end return it (and get rid of the out argument)?